packages/SwingSet/docs/garbage-collection.md

```diff
@@ -64,46 +135,241 @@ When tracking the reachability state of a JavaScript `Object`, we define five st
 * COLLECTED: the JS engine has realized the object is gone
 * FINALIZED: a `FinalizationRegistry` callback is being run
 
-If you had a `WeakRef` for the object, it would be "full" (i.e. `.deref()` returns a value) in the REACHABLE and UNREACHABLE states, and "empty" (`wr.deref() === undefined`) in COLLECTED and FINALIZED.
+If you had a `WeakRef` for the object, it would be "alive" (i.e. `.deref()` returns a value) in the REACHABLE and UNREACHABLE states, and "dead" (`wr.deref() === undefined`) in COLLECTED and FINALIZED.
 
-Note that there's no actual state machine with those values, and we can't observe all of the transitions from JavaScript, but we can describe what operations could cause a transition, and what our observations allow us to deduce about the state:
+Note that there's no actual state machine with those values, and we can't observe all of the transitions from JavaScript. But we *can* describe what operations could cause a transition, and what our observations allow us to deduce about the state:
 
 * UKNOWN moves to REACHABLE when a delivery introduces a new import
   * userspace holds a reference only in REACHABLE
 * REACHABLE moves to UNREACHABLE only during a userspace crank
-* UNREACHABLE moves to COLLECTED when GC runs, which queues the finalizer
+* UNREACHABLE moves to COLLECTED when GC runs, which queues the finalizer callback
 * COLLECTED moves to FINALIZED when a new turn runs the finalizer
-* liveslots moves from FINALIZED to UNKNOWN by syscalling dropImports
+* liveslots moves from FINALIZED to UNKNOWN by doing `syscall.dropImports`
 
 We have several subtle challenges to keep in mind:
 
 * we cannot sense the difference between REACHABLE and UNREACHABLE, although we know it can only happen as userspace runs
-* the transition from UNREACHABLE to COLLECTED can happen spontaneously, at any moment, whenever the engine experience memory pressure and decide to run GC
+* the transition from UNREACHABLE to COLLECTED can happen spontaneously, at any moment, whenever the engine experiences memory pressure and decides to run GC
 * a new delivery might re-import an object that was already on its way out
   * the liveslots `slotToVal` WeakRef will re-use the old Presence if present (REACHABLE or UNREACHABLE)
   * if we're in COLLECTED, then a finalizer callback is already queued, and will run sooner or later, so the callback must not clobber a re-import
```

Comment:
Reimportation messes with your model, because it's kind of like a transition from COLLECTED to REACHABLE.  Except that from the kernel or vat-to-vat perspective, it's the same object being resurrected, going from COLLECTED to REACHABLE, but from the internal-to-the-vat perspective it's an entirely new object going from UNKNOWN1 to REACHABLE.  Since what you're implementing (in fact, what's motivating this model entirely) is the vat-to-vat part, it's easy to overlook that we've got these two perspectives coexisting.

Reply:
Yeah, I kinda need a way to explain what JS gives us (WeakRef `.deref()` results, finalizer callbacks) first, in isolation, to build up the notion that we can't observe certain transitions. Then, in a second section, talk about how this relates to vats (REACHABLE<->UNREACHABLE is a userspace transition, so it can't happen spontaneously after userspace is finished). Then tie it in to imports/exports. I'm definitely handwaving here.